### PR TITLE
Workspace version hackery

### PR DESCRIPTION
**Pull request by nicko7i**
Mostly triggered by a function convertVtrqID that mutated its argument (contrary to ESLint rules).

Also, work on the v2 versions of ICR and storage efficiency.

Also, removed some debug comments from v8 code.

### CODE REVIEW

File: vcnc-server/vcnc-rest/src/routes/v1.js [hunk 1]
```diff
@@ -34,51 +34,38 @@ function latency() {
...
-  } else {
-    jsonMap = [];
-  }
-  const jsonResult = {
-    spec: jsonMap,
+  return {
+    spec:
+      convertMapVtrqID(jsonWs.maps)
+      .map(e => Object.assign({}, e, { local })),
   };
```
Comment (ybegus) [inline vcnc-server/vcnc-rest/src/routes/v1.js:54]: Does it work if jsonWs.maps is empty?

File: vcnc-server/vcnc-rest/src/routes/v2.js [hunk 2]
```diff
@@ -34,68 +34,57 @@ function latency() {
...
-  }
+//  Converts a workspace in PIDL JSON format into a workspace in REST JSON
+//  format.
+//
+function workspace(jsonIn, name) {
+  const jsonWs = JSON.parse(jsonIn.ws);
   const jsonResult = {
-    workspace: jsonWs,
+    workspace: Object.assign({}, jsonWs, { name, maps: convertMapVtrqID(jsonWs.maps) }),
   };
```
Comment (ybegus) [inline vcnc-server/vcnc-rest/src/routes/v2.js:52]: Does it work if jsonWs.maps is empty?

**Review by ybegus — COMMENTED** [review]
This is not quite clear that this code works if jsonWs.maps is empty

File: vcnc-server/vcnc-rest/src/routes/v1.js (hunk 1 above)
Comment (nicko7i) [reply vcnc-server/vcnc-rest/src/routes/v1.js:54]: If it's an empty array, yes.  

[].map(e => e) returns []

If, for example, an evil Boost property tree converted the [] into '', then it wouldn't work.  I added a check as part of PR #17

File: vcnc-server/vcnc-rest/src/routes/v2.js (hunk 2 above)
Comment (nicko7i) [reply vcnc-server/vcnc-rest/src/routes/v2.js:52]: Please see my earlier  comment.

**Pull request merged**